examples/coap-server.c

```diff
@@ -304,6 +305,210 @@ check_async(coap_context_t *ctx,
+    if (coap_string_equal(uri_path, dynamic_entry[i].uri_path)) {
+      /* Dynamic entry no longer required - delete it */
+      coap_delete_string(dynamic_entry[i].value);
+      if (dynamic_count-i > 1) {
+         memmove (&dynamic_entry[i],
+                  &dynamic_entry[i+1],
+                 (dynamic_count-i-1) * sizeof (dynamic_entry[0]));
+      }
+      dynamic_count--;
+      break;
+    } 
+  }
+
+  /* Dynamic resource no longer required - delete it */
+  coap_delete_resource(ctx, resource);
+  response->code = COAP_RESPONSE_CODE(202);
+  return;
+}
+
+/*
+ * Regular GET handler - used by resources created by the
+ * Unknown Resource PUT handler
+ */
+
+static void
+hnd_get(coap_context_t *ctx UNUSED_PARAM,
+        coap_resource_t *resource,
+        coap_session_t *session,
+        coap_pdu_t *request UNUSED_PARAM,
+        coap_string_t *token,
+        coap_string_t *query UNUSED_PARAM,
+        coap_pdu_t *response
+) {
+  coap_str_const_t *uri_path;
+  uint8_t buf[4];
+  int i;
+
+  /*
+   * request will be NULL if an Observe triggered request, so the uri_path,
+   * if needed, must be abstracted from the resource.
+   * The uri_path string is a const pointer
+   */
+
+  uri_path = coap_resource_get_uri_path(resource);
+  if (!uri_path) {
+    response->code = COAP_RESPONSE_CODE(404);
+    return;
+  }
+
+  for (i = 0; i < dynamic_count; i++) {
+    if (coap_string_equal(uri_path, dynamic_entry[i].uri_path)) {
+      break;
+    } 
+  }
+  if (i == dynamic_count) {
+    response->code = COAP_RESPONSE_CODE(404);
+    return;
+  }
+
+  if (coap_find_observer(resource, session, token)) {
+    coap_add_option(response, COAP_OPTION_OBSERVE, coap_encode_var_safe(buf, sizeof (buf), resource->observe), buf);
+  }
+  if (dynamic_entry[i].value) {
+    coap_add_data (response, dynamic_entry[i].value->length, dynamic_entry[i].value->s);
+  }
+  response->code = COAP_RESPONSE_CODE (205);
+}
+
+/*
+ * Regular PUT handler - used by resources created by the
+ * Unknown Resource PUT handler
+ */
+
+static void
+hnd_put(coap_context_t *ctx UNUSED_PARAM,
+        coap_resource_t *resource UNUSED_PARAM,
+        coap_session_t *session UNUSED_PARAM,
+        coap_pdu_t *request,
+        coap_string_t *token UNUSED_PARAM,
+        coap_string_t *query UNUSED_PARAM,
+        coap_pdu_t *response
+) {
+  coap_string_t *uri_path;
+  int i;
+  size_t size;
+  uint8_t *data;
+
+  /* get the uri_path */
+  uri_path = coap_get_uri_path(request);
+  if (!uri_path) {
+    response->code = COAP_RESPONSE_CODE(404);
+    return;
+  }
+
+  for (i = 0; i < dynamic_count; i++) {
+    if (coap_string_equal(uri_path, dynamic_entry[i].uri_path)) {
+      break;
+    } 
+  }
+  if (i == dynamic_count) {
+    if (dynamic_count >= support_dynamic) {
+      response->code = COAP_RESPONSE_CODE(406);
+      return;
+    }
+    dynamic_count++;
+    dynamic_entry = realloc (dynamic_entry, dynamic_count * sizeof(dynamic_entry[0])); 
```

Comment:
In theory, this request could fail, therefore `dynamic_entry` should be checked to be non-NULL before use.

Reply:
Changed to
````
    dynamic_count++;
    dynamic_entry = realloc (dynamic_entry, dynamic_count * sizeof(dynamic_entry[0]));
    if (dynamic_entry) {
      dynamic_entry[i].uri_path = uri_path;
      dynamic_entry[i].value = NULL;
      response->code = COAP_RESPONSE_CODE(201);
    }
    else {
      dynamic_count--;
      response->code = COAP_RESPONSE_CODE(500);
    }
````